### PR TITLE
feat: draft docs for mainnet upgrade

### PR DESCRIPTION
**Pull request by daniel1302**
TODOS:

- [X] Enumerate headers
- [X] Update version references
- [x] Update all the links that are not working
- [X] Check steps in practice(I have tested them once today, but I want to do it one more time, as I was updating docs when met some issues)

### CODE REVIEW

File: docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md [hunk 1]
```diff
@@ -0,0 +1,663 @@
...
+
+#### Data node config
+Important config keys that you need to update: 
+
+- `AutoInitialiseFromNetworkHistory` - We recommend setting it to `false` when you start the network from scratch (e.g. checkpoint restart), or if there is no other data node available.
+- `ChainID` - Make sure it matches the new chain ID for your network.
+- `Admin.Server.SocketPath` - Path for the Unix `sock` file; Ensure parent directory exists. Example may be `<VEGA-NETWORK-HOME>/run/datanode.sock`.
+- `API.CoreNodeIP` - IP of the server where the Vega core node is running. Often, it is `127.0.0.1.` (localhost).
+- `API.CoreNodeGRPCPort` - Port you expose for the Vega core gRPC node. Default: `3002`.
+- `SQLStore.wipe_on_startup` - Defines if the data node removes data from the postgreSQL after the restart. We recommend setting it to `false`.
```
Comment (ettec) [inline docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md:126]: This flag is default false and deprecated, no need to update this

File: docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md [hunk 2]
```diff
@@ -0,0 +1,663 @@
...
+
+Format of `BootstrapPeers`:
+
+```
+"/dns/<DATA-NODE-HOST>/tcp/<DATA-NODE-SWARM-PORT>/ipfs/<DATA-NODE-PEER-ID>"
+```
+
+Example `BootstrapPeers` value:
+
+**Do not use the below bootstrap peers as it is an example from a test network.**
```
Comment (ettec) [inline docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md:155]: Would it be worth mentioning you can hit a peers network history rest endpoint to source peer info? e.g

https://api.n00.testnet.vega.xyz/api/v2/networkhistory

yields:

`{"ipfsAddress":"/ip4/164.92.198.87/tcp/4001/p2p/12D3KooWNiWcT93S3P3eiHqGq4a6feaD2cUfbWw9AxgdVt8RzTHJ","swarmKey":"/key/swarm/psk/1.0.0/\n/base16/\n766567612d6661697267726f756e642d32303233303432363135313900000000","swarmKeySeed":"vega-fairground-202304261519","connectedPeers":["/ip4/159.203.12.201/tcp/4001/p2p/12D3KooWACJuzchZQH8Tz1zNmkGCatgcS2DUoiQnMFaALVMo7DpC","/ip4/139.59.0.80/tcp/4001/p2p/12D3KooWMSaQevxg1JcaFxWTpxMjKw1J13bLVLmoxbeSJ5gpXjRh","/ip4/104.248.42.237/tcp/4001/p2p/12D3KooWDy26YcZVTPbYCmZu3Jukp7Ygjf33ksXTV6GENNowiqBa","/ip4/104.248.42.237/tcp/51338/p2p/12D3KooWDy26YcZVTPbYCmZu3Jukp7Ygjf33ksXTV6GENNowiqBa","/ip4/139.59.0.80/tcp/58318/p2p/12D3KooWMSaQevxg1JcaFxWTpxMjKw1J13bLVLmoxbeSJ5gpXjRh"]}`

the value of ipfsAddress field in the above output can be used as a bootstrap peer

File: docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md [hunk 3]
```diff
@@ -0,0 +1,663 @@
...
+
+- `AutoInitialiseFromNetworkHistory` - We recommend setting it to `false` when you start the network from scratch (e.g. checkpoint restart), or if there is no other data node available.
+- `ChainID` - Make sure it matches the new chain ID for your network.
+- `Admin.Server.SocketPath` - Path for the Unix `sock` file; Ensure parent directory exists. Example may be `<VEGA-NETWORK-HOME>/run/datanode.sock`.
+- `API.CoreNodeIP` - IP of the server where the Vega core node is running. Often, it is `127.0.0.1.` (localhost).
+- `API.CoreNodeGRPCPort` - Port you expose for the Vega core gRPC node. Default: `3002`.
+- `SQLStore.wipe_on_startup` - Defines if the data node removes data from the postgreSQL after the restart. We recommend setting it to `false`.
+- `SQLStore.UseEmbedded` - If true, internal (managed by the binary itself) postgreSQL is used. We strongly recommend setting it to `false` for production, as it is intended for testing only.
+- `SQLStore.ConnectionConfig` - Update the entire section, as it is where you set PostgreSQL credentials.
+- `NetworkHistory.Enabled` - Enables IPFS network history. See the [network history](#a-bit-about-network-history) section below.
```
Comment (ettec) [inline docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md:129]: Do we need to specify that they should update this?  default is true,  I don't see any reason why we would want them to run with it false (?)

File: docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md (hunk 2 above)
Comment (candida-d) [reply docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md:155]: This looks good, but if someone is in the process of upgrading, will this URL work for mainnet? Or do they all need to upgrade before this is available and thus they can only use it after this version?

File: docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md (hunk 1 above)
Comment (candida-d) [reply docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md:126]: I think we should actually advise people to delete this manually if necessary - what do you think @ettec? That way there's no risk of them having something from a previous version>

File: docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md [hunk 4]
```diff
@@ -0,0 +1,663 @@
...
+AmbientCapabilities=CAP_NET_BIND_SERVICE
+
+[Install]
+WantedBy=multi-user.target
+```
+
+
+
+### 16. Install/Upgrade PostgreSQL for data node
+If you are running a data node, you will need to install or upgrade PostgreSQL. The PostgreSQL instance must be on the same server as the data node, as the PostgreSQL does data-snapshot in the data node location file space. The data node uses PostgreSQL snapshots for the network history feature.
```
Comment (ettec) [inline docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md]: `The PostgreSQL instance must be on the same server as the data node, as the PostgreSQL does data-snapshot in the data node location file space`  -  The data snapshot process no longer relies on being in the same location file space, although having said that, afaik we have not tested thoroughly - so we should probably stick with the recommendation for now - but not for the reason given.

File: docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md (hunk 1 above)
Comment (ettec) [reply docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md:126]: yes, that would be better

File: docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md [hunk 5]
```diff
@@ -0,0 +1,657 @@
...
+
+#### Data node config
+Important config keys that you need to update/check: 
+
+- `AutoInitialiseFromNetworkHistory` - We recommend setting it to `false` when you start the network from scratch (e.g. checkpoint restart), or if there is no other data node available.
+- `ChainID` - Make sure it matches the new chain ID for your network.
+- `Admin.Server.SocketPath` - Path for the Unix `sock` file; Ensure parent directory exists. Example may be `<VEGA-NETWORK-HOME>/run/datanode.sock`.
+- `API.CoreNodeIP` - IP of the server where the Vega core node is running. Often, it is `127.0.0.1.` (localhost).
+- `API.CoreNodeGRPCPort` - Port you expose for the Vega core gRPC node. Default: `3002`.
+- `SQLStore.wipe_on_startup` - Defines if the data node removes data from the postgreSQL after the restart. We recommend setting it to `false`.
```
Comment (candida-d) [inline docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md:126]:
```suggestion
- `SQLStore.wipe_on_startup` - Defines if the data node removes data from the postgreSQL after the restart. We recommend deleting this, if it's in your config, as it has been deprecated and is not needed.
```

File: docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md (hunk 2 above)
Comment (candida-d) [reply docs/node-operators/migration-guides/upgrade-mainnet-to-0714.md:155]: @ettec hopefully something we can remember to improve in the docs for a future upgrade

**Review by candida-d — APPROVED** [review]
Waiting on the upgrading file

**Pull request merged**